include/core/shape.h

```
@@ -757,6 +839,191 @@ class CompositeShape : public Shape<dim>
+  {
+    // First, we read the shape file name
+    local_file_name = file_name;
+#ifdef DEAL_II_WITH_OPENCASCADE
+    // Checks the file name extension to identify which type of OpenCascade
+    // shape we are working with.
+    std::vector<std::string> file_name_and_extension(
+      Utilities::split_string_list(local_file_name, "."));
+
+    // Load the shape with the appropriate tool.
+    if (file_name_and_extension[1] == "step" ||
+        file_name_and_extension[1] == "stp")
+      {
+        shape = OpenCASCADE::read_STEP(local_file_name);
+        this->additional_info_on_shape = "step";
+      }
+    else if (file_name_and_extension[1] == "iges" ||
+             file_name_and_extension[1] == "igs")
+      {
+        shape = OpenCASCADE::read_IGES(local_file_name);
+        this->additional_info_on_shape = "iges";
+      }
+    else if (file_name_and_extension[1] == "stl")
+      {
+        shape                          = OpenCASCADE::read_STL(local_file_name);
+        this->additional_info_on_shape = "stl";
+      }
+    else
+      {
+        throw std::runtime_error(
+          "Wrong file extension for an opencascade shape. The possible file type are: step, stp, iges, igs, stl.");
+      }
+
+    // used this local variable as the shape tolerance in the calculations.
+    shape_tol = OpenCASCADE::get_shape_tolerance(shape);
+
+    // Initialize some variables and the OpenCascade distance tool.
+    OpenCASCADE::extract_compound_shapes(
+      shape, compounds, compsolids, solids, shells, wires);
+    vertex_position = OpenCASCADE::point(Point<dim>());
+    vertex          = BRepBuilderAPI_MakeVertex(vertex_position);
+    distancetool    = BRepExtrema_DistShapeShape(shape, vertex);
+
+    // Check if the shape has a shell. If it has a shell, we initialize a
+    // distance tool with just the shell.
+    if (shells.size() > 0)
+      {
+        if (shells.size() > 1)
+          {
+            std::runtime_error(
```

You need to throw the error